src/message/get-log/index.tsx

```diff
@@ -1,154 +1,128 @@
 
-export enum LogParametersTypeProps {
-  remoteLocation = 'remoteLocation',
-  oldestTimestamp = 'oldestTimestamp',
-  latestTimestamp = 'latestTimestamp',
-}
-
-export class LogParametersType {
-  // todo
-  // @Type(() => CustomDataType)
-  // @IsOptional()
-  // customData: CustomDataType | null = null;
-
-  @IsNotEmpty()
-  @IsUrl({ require_tld: false })
-  @CustomFormRender(() => {
-    return (
-      <Form.Item
-        label={LogParametersTypeProps.remoteLocation}
-        name={[GetLogRequestProps.log, LogParametersTypeProps.remoteLocation]}
-        rules={[
-          {
-            required: true,
-            whitespace: true,
-            message: 'Please enter a valid URL.',
-            type: 'url',
-          },
-        ]}
-      >
-        <Input />
-      </Form.Item>
-    );
-  })
-  remoteLocation!: string;
-
-  @TransformDate()
-  @IsOptional()
-  oldestTimestamp?: Date;
-
-  @TransformDate()
-  @IsOptional()
-  latestTimestamp?: Date;
-}
-
-export enum GetLogRequestProps {
-  log = 'log',
-  logType = 'logType',
-  requestId = 'requestId',
-  retries = 'retries',
-  retryInterval = 'retryInterval',
-}
-
-export class GetLogRequest {
-  // todo
-  // @Type(() => CustomDataType)
-  // @IsOptional()
-  // customData: CustomDataType | null = null;
-
-  @Type(() => LogParametersType)
-  @ValidateNested()
-  @IsNotEmpty()
-  log!: LogParametersType;
-
-  @IsEnum(LogEnumType)
-  logType!: LogEnumType;
-
-  @IsInt()
-  requestId!: number;
-
-  @IsInt()
-  @IsOptional()
-  retries?: number;
-
-  @IsInt()
-  @IsOptional()
-  retryInterval?: number;
-}
-
-export class GetLogResponse {
-  // todo
-  // @Type(() => CustomDataType)
-  // @ValidateNested()
-  // @IsOptional()
-  // customData?: CustomDataType;
-
-  @IsEnum(LogStatusEnumType)
-  status!: LogStatusEnumType;
-
-  @Type(() => StatusInfoType)
-  @ValidateNested()
-  @IsOptional()
-  statusInfo?: StatusInfoType;
-}
-
 export interface GetLogProps {
   station: ChargingStation;
 }
 
 export const GetLog: React.FC<GetLogProps> = ({ station }) => {
-  const [formProps] = Form.useForm();
   const formRef = useRef();
-
-  const getLogRequest = new GetLogRequest();
-  getLogRequest[GetLogRequestProps.requestId] = generateRandomSignedInt();
-  getLogRequest[GetLogRequestProps.log] = new LogParametersType();
-  getLogRequest[GetLogRequestProps.log][LogParametersTypeProps.remoteLocation] =
-    `${DIRECTUS_URL}/files`;
-
-  const handleSubmit = useCallback(
-    async (plainValues: any) => {
-      const classInstance = plainToInstance(GetLogRequest, plainValues);
-      await triggerMessageAndHandleResponse(
+  const [form] = Form.useForm();
+  const formProps = { form };
+
+  const [loading, setLoading] = useState<boolean>(false);
+  const [valid, setValid] = useState<boolean>(false);
+  
+  const apiUrl = useApiUrl();
+  const {
+    data: requestIdResponse,
+    isLoading: isLoadingRequestId,
+    // isError: isErrorLoadingRequestId,
+  } = useCustom<any>({
```

Comment:
please note, in this PR https://github.com/citrineos/citrineos-operator-ui/pull/25 we created the `useGqlCustom` helper method which will should more or less simplify this call to just be:
```
const {
    data: requestIdResponse,
    isLoading: isLoadingRequestId,
    // isError: isErrorLoadingRequestId,
  } = useCustom<any>({
      gqlQuery: CHARGING_STATION_SEQUENCES_GET_QUERY,
      variables: {
        stationId: station.id,
        type: 'getLog',
      },
  }
```